src/main/java/ca/corefacility/bioinformatics/irida/config/data/IridaApiJdbcDataSourceConfig.java

```diff
@@ -102,10 +108,29 @@ public SpringLiquibase springLiquibase(final DataSource dataSource, final Applic
 						+ "]");
 			}
 			liquibaseShouldRun = Boolean.FALSE;
+		} else if (fixLiquibaseChangeSetFilenames) {
+			logger.info("Removing 'classpath:' prefix from FILENAME column in DATABASECHANGELOG table.");
+			fixLiquibaseChangeSetFilenames(dataSource);
 		}
 
 		springLiquibase.setShouldRun(liquibaseShouldRun);
 
 		return springLiquibase;
 	}
+
+	/**
+	 * Method to execute sql before liquibase initializes DB to strip `classpath:`
+	 * prefix from FILENAME column in DATABASECHANGELOG table. Fixes issue where
+	 * liquibase 3.5.1 was including prefix when run through spring but not via
+	 * command line. (https://liquibase.jira.com/browse/CORE-2766)
+	 * 
+	 * @param dataSource
+	 * @throws ScriptException
+	 */
+	private void fixLiquibaseChangeSetFilenames(DataSource dataSource) throws ScriptException {
+		ResourceLoader resourceLoader = new DefaultResourceLoader();
+		Resource sqlScript = resourceLoader.getResource("classpath:ca/corefacility/bioinformatics/irida/sql/fix-liquibase-changeset-filenames.sql");
+		ResourceDatabasePopulator populator = new ResourceDatabasePopulator(sqlScript);
+		DatabasePopulatorUtils.execute(populator, dataSource);
```

Comment:
One issue that comes up here is that if you are starting from an entirely new database the table `DATABASECHANGELOG`  will not exist and so this query will fail:

```
Factory method 'springLiquibase' threw exception; nested exception is org.springframework.jdbc.datasource.init.ScriptStatementFailedException:
Failed to execute SQL script statement #1 of class path resource [ca/corefacility/bioinformatics/irida/sql/fix-liquibase-changeset-filenames.sql]: UPDATE DATABASECH
ANGELOG SET FILENAME = TRIM("classpath:" from FILENAME) WHERE FILENAME like "classpath:%"; nested exception is java.sql.SQLSyntaxErrorException: Table 'irida_test.D
ATABASECHANGELOG' doesn't exist
```

Reply:
This same error occurs if you try to run:

```
./run.sh --create-db
```

Reply:
Fixed in be677dc, it shouldn't have even run with `./run.sh --create-db` as that is supposed to use hibernate and not liquibase. I added a check for `liquibaseShouldRun` and moved the updated into a stored procedure so that we can check for table existence before attempting to run the sql.

Reply:
Instead of a stored procedure I could have set `populator.setContinueOnError(true)` but I thought that it wasn't the cleanest solution.

Reply:
Thanks so much. This solution looks great. I had seen `INFORMATION_SCHEMA` before when looking to see how you could check if a table exists. Seems to exist across many database systems, which is good (https://en.wikipedia.org/wiki/Information_schema).

Yeah, `setContinueOnError` isn't the cleanest solution. I like the way you've done it. I still have to test it out.